packages/palette/src/Theme.tsx

```diff
@@ -49,12 +49,21 @@ export const useTheme = <T extends TTheme>() => {
   return { theme }
 }
 
-/** Returns a config specific to the current theme */
+/** Returns a config specific to the current theme. For use in React components */
 export const useThemeConfig = <T, U>({ v2, v3 }: { v2: T; v3: U }): U | T => {
   const { theme = { id: "v2" } } = useTheme()
   return theme.id === "v2" ? v2 : v3
 }
 
+/** Returns a config specific to the current theme. For use in styled-components */
+export const getThemeConfig = <T, U>(
+  props: Record<string, any>,
+  { v2, v3 }: { v2: T; v3: U }
+): U | T => {
+  const { theme = { id: "v2" } } = props
+  return theme.id === "v2" ? v2 : v3
+}
+
```

Comment:
We can probably collapse this duplicate function into `useThemeConfig`, where an additional optional arg takes `props`, pushing it forward in precedence

Reply:
Since useTheme is a hook it both needs to be in a React component and can't be called conditionally. So I'm not sure that'd be possible.